### PR TITLE
audacity-gtk3: re-introduce at 3.0.2

### PR DESCRIPTION
**Pull request by peterhoeg**
###### Motivation for this change

Although upstream recommends the gtk2 version, we previously had a gtk3 version as wall. Re-introduce that.

Also some minor cleanups.

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [X] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by peterhoeg** [comment]
Cc @wlib 

Further to https://github.com/NixOS/nixpkgs/pull/121623

File: pkgs/applications/audio/audacity/default.nix [hunk 1]
```diff
@@ -29,24 +29,26 @@
...
+, at-spi2-core
+, dbus
+, epoxy
+, libXdmcp
+, libXtst
+, libpthreadstubs
+, libselinux
+, libsepol
+, libxkbcommon
+, utillinux
```
Comment (OPNA2608) [inline pkgs/applications/audio/audacity/default.nix:41]: > `anonymous function at <…>/pkgs/applications/audio/audacity/default.nix:1:1 called without required argument 'utillinux'`

https://github.com/NixOS/nixpkgs/blob/ce93c98ce2243a8228419fcfbe97fd87f3c5f99e/pkgs/top-level/aliases.nix#L817

```suggestion
, util-linux
```

same in `buildInputs`.

**Review by OPNA2608 — CHANGES_REQUESTED** [review]

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 121788` at 6a6f32ce run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built successfully:</summary>
  <ul>
    <li>audacity (audacity-gtk2)</li>
    <li>audacity-gtk3</li>
  </ul>
</details>
<details>
  <summary>2 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/build-tools-in-build-inputs.md">build-tools-in-build-inputs</a>

util-linux is a build tool so it likely goes to `nativeBuildInputs`, not `buildInputs`.

Near pkgs/applications/audio/audacity/default.nix:110:3:
```
    |
110 |   buildInputs = [
    |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-patch-comment.md">missing-patch-comment</a>

Please add a comment on the line above, explaining the purpose of this patch.
Near pkgs/applications/audio/audacity/default.nix:74:5:
```
   |
74 |     (fetchpatch {
   |     ^
```
</li>
  </ul>
</details>

**Comment by OPNA2608** [comment]
> Also some minor cleanups.

If I may add to that, could we please make it more explicit that the patch is coming from a PR? If you append `.patch` to a PR commit URL and resolve that instead of copying it straight to your Nix file, then GitHub will change the URL from `…/pull/<PR-number>/commits/<commit-id>` to `.../commit/<commit-id>`. It's hard to associate the resulting link with the PR it came from because GitHub will now just tell you it's "not from this repository".

grafik

A comment with a link to the PR

```nix
# https://github.com/audacity/audacity/pull/831
```

or changing the url to the PR commit

```nix
url = "https://github.com/audacity/audacity/pull/831/commits/007852e51fcbb5f1f359d112f28b8984a604dac6.patch";
```

/ entire PR (includes all PR commits so may change more frequently)

```nix
url = "https://github.com/audacity/audacity/pull/831.patch";
```

would be alot nicer.

**Comment by peterhoeg** [comment]
> If I may add to that, could we please make it more explicit that the patch is coming from a PR?

Absolutely, good point.

> url = "https://github.com/audacity/audacity/pull/831.patch";
> would be alot nicer.

The problem with this is that the contents can change when the PR is updated - a specific commit is nice and explicit.

**Comment by OPNA2608** [comment]
`sqlite` can also be de-vendored with ease and it doesn't look like the JACK we pass in is getting picked up by the vendored portaudio.

```
-- ========== Configuring portaudio ==========
-- Using 'portaudio' local library
-- Found ALSA: /nix/store/2cn2jq3lzas1w5c2n4s1pj5xrzgp97dx-alsa-lib-1.2.4/lib/libasound.so (found version "1.2.4") 
CMake Warning at cmake-proxies/portaudio-v19/CMakeLists.txt:91 (find_package):
  By not providing "Findjack.cmake" in CMAKE_MODULE_PATH this project has
  asked CMake to find a package configuration file provided by "jack", but
  CMake did not find one.

  Could not find a package configuration file provided by "jack" with any of
  the following names:

    jackConfig.cmake
    jack-config.cmake

  Add the installation prefix of "jack" to CMAKE_PREFIX_PATH or set
  "jack_DIR" to a directory containing one of the above files.  If "jack"
  provides a separate development package or SDK, be sure it has been
  installed.
```

**Comment by wlib** [comment]
@peterhoeg sorry about removing it if you needed it, I just did that because it was entirely broken for me on my hidpi monitor and I found the reason upstream, so I assumed that including it was premature. More options are valuable though, so I probably should have left it anyways.

> A comment with a link to the PR

I think this is the best approach, it's simple and obvious, while keeping the exact commit ref

**Comment by peterhoeg** [comment]
> it was entirely broken for me on my hidpi monitor was premature

The -gtk2 version doesn't adjust to my color scheme making it impossible to read anything, whereas the gtk3 version albeit with some fields not being quite properly aligned, but it definitely works better here.

But no problem at all - the fix is just a PR away anyway! ;-)

**Comment by peterhoeg** [comment]
sqlite de-vendoring was done separately. The jack/portaudio mess should be sorted as well, but it's orthogonal to this.

**Pull request merged**